diff -r -u2 a/language/go/resolve.go b/language/go/resolve.go
--- a/language/go/resolve.go	2020-10-02 14:22:18.000000000 -0700
+++ b/language/go/resolve.go	2020-11-17 19:40:59.770648029 -0800
@@ -20,5 +20,4 @@
 	"fmt"
 	"go/build"
-	"log"
 	"path"
 	"regexp"
@@ -80,5 +79,5 @@
 		resolve = ResolveGo
 	}
-	deps, errs := imports.Map(func(imp string) (string, error) {
+	deps, _ := imports.Map(func(imp string) (string, error) {
 		l, err := resolve(c, ix, rc, imp, from)
 		if err == skipImportError {
@@ -95,7 +94,4 @@
 		return l.String(), nil
 	})
-	for _, err := range errs {
-		log.Print(err)
-	}
 	if !deps.IsEmpty() {
 		if r.Kind() == "go_proto_library" {
