diff --git a/bazel/grpc_extra_deps.bzl b/bazel/grpc_extra_deps.bzl
index 6ac6093a5f..b36e488d88 100644
--- a/bazel/grpc_extra_deps.bzl
+++ b/bazel/grpc_extra_deps.bzl
@@ -56,14 +56,6 @@ def grpc_extra_deps(ignore_version_differences = False):
 
     api_dependencies()
 
-    go_rules_dependencies()
-    go_register_toolchains(version = "1.22.5")
-    gazelle_dependencies()
-
-    # Pull-in the go 3rd party dependencies for protoc_gen_validate, which is
-    # needed for building C++ xDS protos
-    go_third_party()
-
     apple_rules_dependencies(ignore_version_differences = ignore_version_differences)
 
     apple_support_dependencies()
