diff --git a/bazel/grpc_extra_deps.bzl b/bazel/grpc_extra_deps.bzl
index 1acc8b84e8..8240c69618 100644
--- a/bazel/grpc_extra_deps.bzl
+++ b/bazel/grpc_extra_deps.bzl
@@ -23,7 +23,6 @@ load("@com_google_protobuf//:protobuf_deps.bzl", "protobuf_deps")
 load("@envoy_api//bazel:repositories.bzl", "api_dependencies")
 load("@google_cloud_cpp//bazel:google_cloud_cpp_deps.bzl", "google_cloud_cpp_deps")
 load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_dependencies")
-load("@rules_java//java:rules_java_deps.bzl", "rules_java_dependencies")
 load("@rules_proto//proto:repositories.bzl", "rules_proto_dependencies")
 load("@rules_python//python:repositories.bzl", "py_repositories")
 load("@rules_shell//shell:repositories.bzl", "rules_shell_dependencies", "rules_shell_toolchains")
@@ -54,8 +53,6 @@ def grpc_extra_deps(ignore_version_differences = False):
     rules_shell_dependencies()
     rules_shell_toolchains()
 
-    rules_java_dependencies()
-
     protobuf_deps()
 
     rules_proto_dependencies()
@@ -63,14 +60,6 @@ def grpc_extra_deps(ignore_version_differences = False):
 
     api_dependencies()
 
-    go_rules_dependencies()
-    go_register_toolchains(version = "1.22.5")
-    gazelle_dependencies()
-
-    # Pull-in the go 3rd party dependencies for protoc_gen_validate, which is
-    # needed for building C++ xDS protos
-    go_third_party()
-
     apple_rules_dependencies(ignore_version_differences = ignore_version_differences)
 
     apple_support_dependencies()
