diff --git a/go/private/rules/test.bzl b/go/private/rules/test.bzl
index 12dd1cc5..dcb4eb63 100644
--- a/go/private/rules/test.bzl
+++ b/go/private/rules/test.bzl
@@ -150,9 +150,6 @@ def _go_test_impl(ctx):
     )
 
     test_gc_linkopts = gc_linkopts(ctx)
-    if not go.mode.debug and go.mode.strip:
-        # Disable symbol table and DWARF generation for test binaries.
-        test_gc_linkopts.extend(["-s", "-w"])
 
     # Link in the run_dir global for bzltestutil.
     # We add "+initfirst/" to the package path so the package is initialized
