diff --git a/go/private/platforms.bzl b/go/private/platforms.bzl
index 664f7aed..c0267648 100644
--- a/go/private/platforms.bzl
+++ b/go/private/platforms.bzl
@@ -32,6 +32,7 @@ BAZEL_GOARCH_CONSTRAINTS = {
     "arm64": "@platforms//cpu:aarch64",
     "ppc64": "@platforms//cpu:ppc",
     "ppc64le": "@platforms//cpu:ppc",
+    "riscv64": "@platforms//cpu:riscv64",
     "s390x": "@platforms//cpu:s390x",
 }

